tests/test_device.py

```diff
@@ -15,24 +15,37 @@
 PAGE_SIZE = resource.getpagesize()
 
 
-class DeviceTest(unittest.TestCase):
+class DeviceTest(PyverbsAPITestCase):
     """
     Test various functionalities of the Device class.
     """
 
-    def test_dev_list(self):
-        """
-        Verify that it's possible to get IB devices list.
-        """
-        d.get_device_list()
+    def setUp(self):
+        pass
 
-    @staticmethod
-    def get_device_list():
+    def tearDown(self):
+        pass
+
+    def get_device_list(self):
         lst = d.get_device_list()
         if len(lst) == 0:
             raise unittest.SkipTest('No IB device found')
+        dev_name = self.config['dev']
```

Comment:
This is a weird behavior IMO. Why would a get_device_list function get a device name parameter? I would expect it to return a list of all devices.

Reply:
The idea is to run the tests on a specific device on a system with multiple devices if the user passes a '--dev' parameter, otherwise the return value will be a list of all devices in the system.

Reply:
Test cases that do not use the dev_name and other fixed parameters should be inheriting from a different class.

These are the 'old style' tests that should be deleted or updated..